File: apk_monytix/app/src/main/java/com/example/apk/console/tabs/AIInsightTabView.kt
package com.example.apk.console.tabs

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.example.apk.console.MolyConsoleViewModel
import com.example.apk.ui.components.GlassCard
import com.example.apk.ui.theme.Gold
import com.example.apk.ui.theme.TextPrimary
import com.example.apk.ui.theme.TextSecondary

@Composable
fun AIInsightTabView(viewModel: MolyConsoleViewModel) {
    val insights by viewModel.aiInsights.collectAsStateWithLifecycle()
    val isLoading by viewModel.isInsightsLoading.collectAsStateWithLifecycle()
    val error by viewModel.insightsError.collectAsStateWithLifecycle()
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        // AI Banner
        AIBanner()
        
        when {
            isLoading -> {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator(color = Gold)
                }
            }
            error != null -> {
                val errorMsg = error
                InsightsErrorState(error = errorMsg ?: "Unknown error") {
                    viewModel.loadAIInsights()
                }
            }
            insights.isEmpty() -> {
                InsightsEmptyState()
            }
            else -> {
                InsightsContent(insights = insights)
            }
        }
    }
}

@Composable
fun AIBanner() {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(
                Brush.horizontalGradient(
                    colors = listOf(
                        Color(0xFF8B5CF6),
                        Color(0xFF8B5CF6).copy(alpha = 0.8f)
                    )
                )
            )
            .padding(16.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = Icons.Default.AutoAwesome,
            contentDescription = null,
            tint = Color.White,
            modifier = Modifier.size(20.dp)
        )
        
        Column(
            verticalArrangement = Arrangement.spacedBy(4.dp),
            modifier = Modifier.weight(1f)
        ) {
            Text(
                text = "Molytix AI",
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White
            )
            
            Text(
                text = "Personalized insights and recommendations based on your financial behavior.",
                fontSize = 12.sp,
                color = Color.White.copy(alpha = 0.9f),
                maxLines = 2
            )
        }
    }
}

@Composable
fun InsightsContent(insights: List<com.example.apk.console.models.AIInsight>) {
    Column(
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        Text(
            text = "Today's Insights",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary,
            modifier = Modifier.fillMaxWidth()
        )
        
        insights.forEach { insight ->
            InsightCard(insight = insight)
        }
    }
}

@Composable
fun InsightCard(insight: com.example.apk.console.models.AIInsight) {
    GlassCard(
        padding = PaddingValues(20.dp),
        cornerRadius = 16.dp
    ) {
        Column(
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(
                    imageVector = getInsightIcon(insight.type),
                    contentDescription = null,
                    tint = getInsightColor(insight.type),
                    modifier = Modifier.size(20.dp)
                )
                
                Text(
                    text = insight.title,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    color = TextPrimary
                )
            }
            
            Text(
                text = insight.message,
                fontSize = 15.sp,
                color = TextPrimary.copy(alpha = 0.9f)
            )
        }
    }
}

@Composable
private fun InsightsErrorState(error: String, onRetry: () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Icon(
            imageVector = Icons.Default.Error,
            contentDescription = null,
            tint = Color(0xFFE53935).copy(alpha = 0.7f),
            modifier = Modifier.size(48.dp)
        )
        
        Text(
            text = "Unable to Load Insights",
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary
        )
        
        Text(
            text = error,
            fontSize = 14.sp,
            color = TextSecondary,
            modifier = Modifier.padding(horizontal = 32.dp)
        )
        
        Button(
            onClick = onRetry,
            colors = ButtonDefaults.buttonColors(containerColor = Gold)
        ) {
            Text("Retry", color = com.example.apk.ui.theme.Charcoal)
        }
    }
}

@Composable
private fun InsightsEmptyState() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Icon(
            imageVector = Icons.Default.AutoAwesome,
            contentDescription = null,
            tint = TextSecondary.copy(alpha = 0.5f),
            modifier = Modifier.size(48.dp)
        )
        
        Text(
            text = "No Insights Yet",
            fontSize = 18.sp,
            fontWeight = FontWeight.SemiBold,
            color = TextPrimary
        )
        
        Text(
            text = "AI insights will appear here as you use the app",
            fontSize = 14.sp,
            color = TextSecondary,
            modifier = Modifier.padding(horizontal = 32.dp)
        )
    }
}

private fun getInsightIcon(type: com.example.apk.console.models.InsightType): androidx.compose.ui.graphics.vector.ImageVector {
    return when (type) {
        com.example.apk.console.models.InsightType.SPENDING_ALERT -> Icons.Default.Warning
        com.example.apk.console.models.InsightType.GOAL_PROGRESS -> Icons.Default.CheckCircle
        com.example.apk.console.models.InsightType.INVESTMENT_RECOMMENDATION -> Icons.Default.TrendingUp
        com.example.apk.console.models.InsightType.BUDGET_TIP -> Icons.Default.Lightbulb
        com.example.apk.console.models.InsightType.SAVINGS_OPPORTUNITY -> Icons.Default.AccountBalance
    }
}

private fun getInsightColor(type: com.example.apk.console.models.InsightType): Color {
    return when (type) {
        com.example.apk.console.models.InsightType.SPENDING_ALERT -> Color(0xFFFFEB3B)
        com.example.apk.console.models.InsightType.GOAL_PROGRESS -> Color(0xFF4CAF50)
        com.example.apk.console.models.InsightType.INVESTMENT_RECOMMENDATION -> Color(0xFF9C27B0)
        com.example.apk.console.models.InsightType.BUDGET_TIP -> Color(0xFF2196F3)
        com.example.apk.console.models.InsightType.SAVINGS_OPPORTUNITY -> Color(0xFF4CAF50)
    }
}

File: ios_monytix/ios_monytix/console/AIInsightTabView.swift
//
//  AIInsightTabView.swift
//  ios_monytix
//
//  Created by santosh on 05/01/26.
//

import SwiftUI

struct AIInsightTabView: View {
    @ObservedObject var viewModel: MolyConsoleViewModel
    
    private let goldColor = Color(red: 0.831, green: 0.686, blue: 0.216)
    private let purpleColor = Color(red: 0.545, green: 0.361, blue: 0.965) // #8B5CF6
    
    var body: some View {
        ScrollView {
            VStack(spacing: 20) {
                // Molytix AI Banner
                aiBanner
                
                if viewModel.isInsightsLoading {
                    ProgressView()
                        .tint(goldColor)
                        .padding(.top, 20)
                } else if let error = viewModel.insightsError {
                    errorState(error)
                } else if viewModel.aiInsights.isEmpty {
                    emptyState
                } else {
                    insightsContent
                }
            }
            .padding(20)
        }
        .refreshable {
            await viewModel.loadAIInsights()
        }
    }
    
    private var aiBanner: some View {
        HStack(spacing: 12) {
            Image(systemName: "sparkles")
                .font(.system(size: 20, weight: .semibold))
                .foregroundColor(.white)
            
            VStack(alignment: .leading, spacing: 4) {
                Text("Molytix AI")
                    .font(.system(size: 16, weight: .bold))
                    .foregroundColor(.white)
                
                Text("Personalized insights and recommendations based on your financial behavior.")
                    .font(.system(size: 12))
                    .foregroundColor(.white.opacity(0.9))
                    .lineLimit(2)
            }
            
            Spacer()
        }
        .padding(16)
        .background(
            LinearGradient(
                colors: [purpleColor, purpleColor.opacity(0.8)],
                startPoint: .leading,
                endPoint: .trailing
            )
        )
        .cornerRadius(12)
    }
    
    private var insightsContent: some View {
        VStack(alignment: .leading, spacing: 20) {
            Text("Today's Insights")
                .font(.system(size: 20, weight: .bold))
                .foregroundColor(.white)
                .frame(maxWidth: .infinity, alignment: .leading)
            
            ForEach(viewModel.aiInsights) { insight in
                InsightCard(insight: insight)
            }
        }
    }
    
    private func errorState(_ error: String) -> some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 48))
                .foregroundColor(.red.opacity(0.7))
            
            Text("Unable to Load Insights")
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            
            Text(error)
                .font(.system(size: 14))
                .foregroundColor(.gray.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            
            Button("Retry") {
                Task {
                    await viewModel.loadAIInsights()
                }
            }
            .buttonStyle(GoldButtonStyle())
            .padding(.top, 8)
        }
        .padding(.top, 40)
    }
    
    private var emptyState: some View {
        VStack(spacing: 16) {
            Image(systemName: "sparkles")
                .font(.system(size: 48))
                .foregroundColor(.gray.opacity(0.5))
            
            Text("No Insights Yet")
                .font(.system(size: 18, weight: .semibold))
                .foregroundColor(.white)
            
            Text("AI insights will appear here as you use the app")
                .font(.system(size: 14))
                .foregroundColor(.gray.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
        }
        .padding(.top, 40)
    }
}

// MARK: - Insight Card

struct InsightCard: View {
    let insight: AIInsight
    
    private let goldColor = Color(red: 0.831, green: 0.686, blue: 0.216)
    
    var body: some View {
        GlassCard(padding: 20, cornerRadius: 16) {
            VStack(alignment: .leading, spacing: 12) {
                HStack(spacing: 12) {
                    Image(systemName: insight.type.icon)
                        .font(.system(size: 20, weight: .semibold))
                        .foregroundColor(colorForType(insight.type))
                    
                    Text(insight.title)
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(.white)
                    
                    Spacer()
                }
                
                Text(insight.message)
                    .font(.system(size: 15))
                    .foregroundColor(.white.opacity(0.9))
                    .lineLimit(nil)
            }
        }
    }
    
    private func colorForType(_ type: InsightType) -> Color {
        switch type {
        case .spendingAlert:
            return .yellow
        case .goalProgress:
            return .green
        case .investmentRecommendation:
            return .purple
        case .budgetTip:
            return .blue
        case .savingsOpportunity:
            return .green
        }
    }
}

#Preview {
    let authService = AuthService()
    return AIInsightTabView(viewModel: MolyConsoleViewModel(authService: authService))
        .background(Color(red: 0.18, green: 0.18, blue: 0.18))
}

